### ISSUE TITLE
Handle br tag in flowchart the same way as in sequenceDiagram (especially when exporting to SVG)

### ISSUE THREAD

**Issue by etimau** [mermaid-js/mermaid#1111]
When using `<br>` in a sequenceDiagram, when exported to SVG, it is translated to `<text></text>` tags.
See [this example](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG5BLT4-QjogYSB2ZXJ5IHNpbXBsZSBtdWx0aTxici8-bGluZSBzZW50ZW5jZSIsIm1lcm1haWQiOnsidGhlbWUiOiJuZXV0cmFsIn19)

Is it possible to have the same behavior in flowcharts?
If you try [this](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVEQ7XG5BIC0tXCJtdWx0aTxicj5saW5lXCItLT4gQiIsIm1lcm1haWQiOnsidGhlbWUiOiJuZXV0cmFsIn19) and click on "Download SVG", the result has a parse XML error because of the `<br>` tag.
Idem if the `<br>` is inside of a node.

**Comment by GDFaber** [comment]
I can take a look here, because I'm also interested in #702 which also is about line breaks in sequence diagrams.

**Comment by chris579** [comment]
> I'm also interested in #702 which is about line breaks in sequence diagrams

So this is a duplicate then?

**Comment by chris579** [comment]
> I can take a look here, because I'm also interested in #702 which is about line breaks in sequence diagrams.

Great! Make sure to make two separate prs for these issues.

**Comment by GDFaber** [comment]
This can be solved by turning ```<br>``` into ```<br/>``` to have all tags closed in the label text. I will work on this after #1156 has been merged (because it modifies the regex for line breaks in flowRenderer.js and it seems like I need to modify the same code block as well).

**Comment by GDFaber** [comment]
> This can be solved by turning ```<br>``` into ```<br/>``` to have all tags closed in the label text.

This won't do much because browsers (at least Chrome) tend to turn ```<br/>``` back into ```<br>``` in order to produce HTML instead of XHTML (if not explicitly set by doctype).

I also tried to wrapping the texts in div-Blocks and replacing the line breaks by ```</div><div>```. This works for security level "strict" but has a drawback in case of "loose": If the html text already includes its own div-Blocks, they might get broken by this approach. I think it cannot be done that way either.

**Comment by GDFaber** [comment]
Turns out, it already works if ```htmlLabels``` is set to ```false``` in the [flowchart configuration](https://mermaid-js.github.io/mermaid/#/mermaidAPI?id=configuration). That's because the text lines get split up into ```<text>``` blocks just as @etimau suggested in the first place.

Is that good enough to work with, or do we need more here?

**Comment by etimau** [comment]
Hi and thanks for you answer and best wishes for 2020!

Maybe I'm doing it wrong but, when I turn htmlLabels to false, I don't get a line breaks but I _see_ the `<br/>` tag in the text...

Furthermore, the rendering is a bit messy.
Here is the rendering with htmlLabels set to true:
![htmlLabelsTrue](https://user-images.githubusercontent.com/319380/71673435-1bdd4480-2d79-11ea-817c-f5667b98c8d5.png)
(You can also [see it here](https://mermaidjs.github.io/mermaid-live-editor/#/view/eyJjb2RlIjoiZ3JhcGggVERcbmxpbmtTdHlsZSBkZWZhdWx0IGludGVycG9sYXRlIG1vbm90b25lWVxuXG4lJWNsYXNzRGVmXG5jbGFzc0RlZiBjbGFzc1VQICAgICAgZmlsbDpncmV5LCAgICAgICAgICAgICAgICBzdHJva2Utd2lkdGg6MnB4LCBzdHJva2U6YmxhY2s7XG5jbGFzc0RlZiBjbGFzc0FCT1JURUQgZmlsbDpyZWQsICAgICBjb2xvcjp3aGl0ZSxzdHJva2Utd2lkdGg6MnB4LCBzdHJva2U6ZGFya1JlZDtcbmNsYXNzRGVmIGNsYXNzQU5PT0sgICBmaWxsOmdyZWVuLCAgIGNvbG9yOndoaXRlLHN0b2tlLXdpZHRoOjJweCwgIHN0cm9rZTpkYXJrR3JlZW47XG5jbGFzc0RlZiBjbGFzc0lOVCAgICAgZmlsbDpvcmFuZ2UsICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICBzdHJva2U6ZGFya09yYW5nZSwgb3BhY2l0eTowLjc1O1xuY2xhc3NEZWYgY2xhc3NURU1QICAgIGZpbGw6eWVsbG93LCAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgc3Ryb2tlOmdvbGQsICAgICAgIG9wYWNpdHk6MC41O1xuY2xhc3NEZWYgY2xhc3NFTFcgICAgIGZpbGw6YmxhY2ssICAgY29sb3I6IzAwRkYwMCwgICAgICAgICAgICAgICAgc3Ryb2tlOmdyZXk7XG5jbGFzc0RlZiBjbGFzc1NFQ1UgICAgZmlsbDojRjBGRkYwLCAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICBzdHJva2U6Z3JlZW47XG5jbGFzc0RlZiBjbGFzc1NZU09VVCAgZmlsbDpibGFjaywgICBjb2xvcjp3aGl0ZSwgICAgICAgICAgICAgICAgICBzdHJva2U6Z3JleTtcbmNsYXNzRGVmIGNsYXNzQ0ZUICAgICBmaWxsOmxpZ2h0U2VhR3JlZW4sICAgICAgICAgICAgICAgICAgICAgICAgIHN0cm9rZTpkYXJrQ3lhbjtcbmNsYXNzRGVmIGNsYXNzT0NBICAgICBmaWxsOnllbGxvd0dyZWVuLCAgICAgICAgICAgICAgICAgICAgICAgICAgIHN0cm9rZTpkYXJrR3JlZW4sICBvcGFjaXR5OjAuNTtcbmNsYXNzRGVmIGNsYXNzSk9VICAgICBmaWxsOnBpbmssICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIHN0cm9rZTpkZWVwUGluaztcbmNsYXNzRGVmIGNsYXNzT1VUSUxTICBmaWxsOmxpZ2h0R3JleTtcbmNsYXNzRGVmIGNsYXNzUERGICAgICBmaWxsOndoaXRlLCAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIHN0cm9rZTpsaWdodEdyZXk7XG5jbGFzc0RlZiBjbGFzc1BBUkFNICAgZmlsbDpncmV5LCAgICBjb2xvcjojMDAwMEZGLCAgICAgICAgICAgICAgICBzdHJva2U6bmF2eTtcbmNsYXNzRGVmIGNsYXNzVElUUkUgICBmaWxsOm5vbmUsICAgICAgICAgICAgICAgc3Ryb2tlLXdpZHRoOjBweCwgc3Ryb2tlOmdyZXk7XG5jbGFzc0RlZiBjbGFzc0lPICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIHN0cm9rZS13aWR0aDo1cHg7XG5USVRSRVtcIlRJVExFXCJdO1xuUEFSQU1bXCJQQVJBTSA6PGJyPiYxPVBBUkFNMTxicj4mMj1QQVJBTTI8YnI-JjM9UEFSQU0zPGJyPiY0PVBBUkFNNDxicj5cIl1cblxuLmZpYzEoLmZpYzEpIC0tPiBVUDFcblVQMSAtLT4gLmZpYzIoLmZpYzIpXG5cbi5maWMyKC5maWMyKSAtLSBcInNvcnQ8YnI-S0VZPSgxLDYpXCIgLS0-IC5maWMyKC5maWMyKVxuXG4uZmljMiguZmljMikgLS0-IFVQMUJcbi5maWMzKC5maWMzKSAtLT4gVVAxQlxuLmZpYzQoLmZpYzQpIC0tPiBVUDFCXG5VUDFCIC0tPiAuZmljNG9VUDFiKC5maWM0KVxuVVAxQiAtLT4gLmZpYzUoLmZpYzUpXG5cbi5maWMyKC5maWMyKSAtLT4gVVAxQkVcbi5maWMzKC5maWMzKSAtLT4gVVAxQkVcbi5maWM2KC5maWM2KSAtLT4gVVAxQkVcblVQMUJFIC0tPiAuZmljNm9VUDFiZSguZmljNilcblVQMUJFIC0tPiAuZmljNyguZmljNylcblxuLmZpYzcoLmZpYzcpIC0tIFwiY3JlYXRlPGJyPkFSUkFOR0U9KDEsMSkgJ1hYWDEyMDAwMEEwMDMwMDAnICgyLDEzMilcIiAtLT4gUkVQT1JUMVsvUkVQT1JUMS9dXG5cblJFUE9SVDFbL1JFUE9SVDEvXSAtLSBzZW5kIC0tPiBTRU5EMT5TRU5EIERFVEFJTFNdXG4uZmljMSguZmljMSkgLS0-IFVQMUNcblVQMUMgLS0-IC5maWM4KC5maWM4KVxuXG4uZmljOCguZmljOCkgLS0gXCJzb3J0PGJyPktFWT0xLDYgNyw5XCIgLS0-IC5maWM4KC5maWM4KVxuXG4uZmljOCguZmljOCkgLS0-IFVQMURcbi5maWM5KC5maWM5KSAtLT4gVVAxRFxuVVAxRCAtLT4gLmZpYzEwKC5maWMxMClcblxuLmZpYzgoLmZpYzgpIC0tPiBVUDFERVxuLmZpYzkoLmZpYzkpIC0tPiBVUDFERVxuVVAxREUgLS0-IC5maWMxMSguZmljMTEpXG5VUDFERSAtLT4gLlJFUE9SVDJbL1JFUE9SVDIvXVxuXG4uZmljMTEoLmZpYzExKSAtLSBcImNyZWF0ZTxicj5BUlJBTkdFPSgxLDEpICdYWFgxMjAwMDBBMDA0MDAwJyAoMiwxMzIpXCIgLS0-IFJFUE9SVDNbL1JFUE9SVDMvXVxuXG5SRVBPUlQzWy9SRVBPUlQzL10gLS0gc2VuZCAtLT4gU0VORDI-U0VORCBERVRBSUxTXVxuLmZpYzEoLmZpYzEpIC0tPiBVUDFFXG5VUDFFIC0tPiAuZmljMW9VUDFlKC5maWMxKVxuVVAxRSAtLT4gLmZpYzEyKC5maWMxMilcblxuLmZpYzEyKC5maWMxMikgLS0-IFVQMUdcblVQMUcgLS0-IC5maWMxMyguZmljMTMpXG5cbi5maWMxMyguZmljMTMpIC0tIFwiY3JlYXRlPGJyPkFSUkFOR0U9MSwxICdYWFgxMjAwMDBBMDJEMDAwJyAyLDEzMlwiIC0tPiAuZmljMTQoLmZpYzE0KVxuXG4uZmljMTIoLmZpYzEyKSAtLSBcInNvcnQ8YnI-S0VZPSgxMiw1MClcIiAtLT4gLmZpYzEydCguZmljMTJ0KVxuXG4uZmljMTJ0KC5maWMxMnQpIC0tPiBVUDFHXG5VUDFHIC0tPiAuZmljMTMyKC5maWMxMzIpXG5cbi5maWMxMzIoLmZpYzEzMikgLS0gXCJjcmVhdGU8YnI-QVJSQU5HRT0xLDEgJ1hYWDEyMDAwMEEwMk4wMDAnIDIsMTMyXCIgLS0-IC5maWMxNSguZmljMTUpXG5cblRJVFJFIC0uLSBQQVJBTTtcblBBUkFNIC0uLSAuZmljMSguZmljMSk7XG4lJVN0eWxlXG5jbGFzcyAuZmljMSwuZmljMywuZmljNCwuZmljNG9VUDFiLC5maWM1LC5maWM2LC5maWM2b1VQMWJlLC5maWM5LC5maWMxMCwuZmljMW9VUDFlIGNsYXNzRUxXO1xuY2xhc3MgVVAxLFVQMUIsVVAxQkUsVVAxQyxVUDFELFVQMURFLFVQMUUsVVAxRyBjbGFzc1VQO1xuY2xhc3MgLmZpYzQsLmZpYzRvVVAxYiwuZmljNiwuZmljNm9VUDFiZSwuZmljMSwuZmljMW9VUDFlIGNsYXNzSU87XG5jbGFzcyBSRVBPUlQxLFJFUE9SVDIsUkVQT1JUMyBjbGFzc1BERjtcbmNsYXNzIEFOTzQ0S08gY2xhc3NBQk9SVEVEO1xuY2xhc3MgQU5PNDRPSyBjbGFzc0FOT09LO1xuY2xhc3MgUEFSQU0gY2xhc3NQQVJBTTtcbmNsYXNzIFRJVFJFIGNsYXNzVElUUkU7IiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQifX0))

Here is the same graph when htmlLabels is set to false:
![htmlLabelsFalse](https://user-images.githubusercontent.com/319380/71673507-47602f00-2d79-11ea-93bd-b21955f505c5.png)

**Comment by GDFaber** [comment]
Hi, I also wish you all the best for 2020 :)

Thank you for posting the example, and pardon me, I forgot to say that I tried this on a yet unreleased state of the flowchart renderer which will be included in an upcoming release.

I built from the develop branch of the mermaid repository, which produces the following output for your example graph:

![2020-01-02 16_19_28-Mermaid Quick Test Page](https://user-images.githubusercontent.com/1912783/71674604-ef76f780-2d7b-11ea-854c-2b4e9245b25f.png)

**Comment by GDFaber** [comment]
The issues with style classes which you mentioned (for example in node ".fic1") may be connected to issue #487 as there is currently no support to change the text color by mermaid style classes, and setting ```color``` unfortunately only works with HTML labels activated (which is because there is no ```color``` attribute in SVG to style a ```text``` node; instead you would have to use ```fill``` on the text tag, which is not possible at the moment).

**Comment by etimau** [comment]
Thanks for those informations. I will check when the new version will be available.
Just one detail: the text over the links is not centered anymore in the new renderer version. Will there be a way to center it?

**Comment by GDFaber** [comment]
> the text over the links is not centered anymore

At the moment multiline texts are not centered if html labels are disabled. Thanks for pointing that out, I have opened a new issue to get the texts centered for non-html labels as well: #1177

**Comment by GDFaber** [comment]
Hi @retsyo, feel free to ask if you need further information on this topic.
If you feel like your questions have been answered, would you agree to close this issue?

**Comment by GDFaber** [comment]
I think we can close this issue, feel free to open a new one if you run into problems.
